### PR TITLE
Update register_func() in context.py

### PR DESCRIPTION
**Pull request by DaceT**
Fixes dask-contrib/dask-sql#270

### CODE REVIEW

File: dask_sql/context.py [hunk 1]
```diff
@@ -274,8 +274,10 @@ def register_function(
             SELECT f(x)
             FROM df
 
-        Please note that you can always only have one function with the same name;
-        no matter if it is an aggregation or scalar function.
+        Please keep in mind that you can only have one function with the same name,
+        regardless of whether it is an aggregation or a scalar function. However,
+        if you register two functions with the same name, the last one registered
+        will take precedence over the first one.
```
Comment (charlesbluca) [inline dask_sql/context.py:280]: Should we specify here that the precedence only applies if `replace=True`?

```
Please keep in mind that you can only have one function with the same name,
regardless of whether it is an aggregation or a scalar function. By default,
attempting to register two functions with the same name will raise an error;
setting `replace=True` will give precedence to the most recently registered
function.
```

File: dask_sql/context.py [hunk 2]
```diff
@@ -298,13 +300,30 @@ def f(x):
...
+
         Args:
             f (:obj:`Callable`): The function to register
             name (:obj:`str`): Under which name should the new function be addressable in SQL
             parameters (:obj:`List[Tuple[str, type]]`): A list ot tuples of parameter name and parameter type.
                 Use `numpy dtypes <https://numpy.org/doc/stable/reference/arrays.dtypes.html>`_ if possible.
             return_type (:obj:`type`): The return type of the function
-            replace (:obj:`bool`): Do not raise an error if the function is already present
+            replace (:obj:`bool`): Raises an error if the function is already present. If you want to overwrite
+            a function, set the value to True; otherwise, the default value is False.
```
Comment (charlesbluca) [inline dask_sql/context.py:326]: I think the original arg description (do not raise an error if the function is already present) is still correct here; I think we might be better off here with something like:

```
replace (:obj:`bool`): If `True`, do not raise an error if a function with the same name is already
present; instead, replace the original function. Default is `False`.
```

File: dask_sql/context.py [hunk 3]
```diff
@@ -829,7 +848,7 @@ def _register_callable(
 
             elif schema.functions[lower_name] != f:
                 raise ValueError(
-                    "Registering different functions with the same name is not allowed"
+                    'Registering multiple functions with the same name is only permitted when the parameter "replace=True" is used.'
```
Comment (charlesbluca) [inline dask_sql/context.py:851]: Think we should be good without quotes around the keyword argument here:

```suggestion
                    "Registering multiple functions with the same name is only permitted if replace=True"
```

File: dask_sql/context.py (hunk 1 above)
Comment (DaceT) [reply dask_sql/context.py:280]: > Should we specify here that the precedence only applies if `replace=True`?
> 
> ```
> Please keep in mind that you can only have one function with the same name,
> regardless of whether it is an aggregation or a scalar function. By default,
> attempting to register two functions with the same name will raise an error;
> setting `replace=True` will give precedence to the most recently registered
> function.
> ```

Yes, we should; thanks for catching that!

**Review by charlesbluca — APPROVED** [review]
LGTM, thanks @DaceT!

**Pull request merged**